airflow/providers/amazon/aws/transfers/gcs_to_s3.py

```diff
@@ -183,7 +183,7 @@ def execute(self, context):
                 self.log.info("Saving file to %s", dest_key)
 
                 s3_hook.load_bytes(
-                    file_bytes, key=dest_key, replace=self.replace, acl_policy=self.s3_acl_policy
+                    file_bytes.encode(), key=dest_key, replace=self.replace, acl_policy=self.s3_acl_policy
```

Comment:
The return type of GCSHook.download() is currently `str` but it might be `Union[str, bytes]`. When `filename` is set, it returns `filename` in another case it returns content of downloaded file as `bytes`. So I'll remove `encode()` from here and modify the return type of GCSHook.download().

https://github.com/apache/airflow/blob/master/airflow/providers/google/cloud/hooks/gcs.py#L263